sound/soc/sof/ipc4.c

```diff
@@ -631,11 +631,15 @@ static void sof_ipc4_rx_msg(struct snd_sof_dev *sdev)
 			return;
 
 		ipc4_msg->data_size = data_size;
-		snd_sof_ipc_msg_data(sdev, NULL, ipc4_msg->data_ptr, ipc4_msg->data_size);
+		err = snd_sof_ipc_msg_data(sdev, NULL, ipc4_msg->data_ptr, ipc4_msg->data_size);
+		if (err < 0) {
+			dev_warn(sdev->dev, "failed to read IPC reply data: %d\n", err);
```

Comment:
this might be actually an error here?

Reply:
and the text is incorrect, this is not a reply, this is a notification.

Reply:
> and the text is incorrect, this is not a reply, this is a notification.

ack

Reply:
> this might be actually an error here?

I changed it to dev_err. To be honestly, I am not sure what is the impact if we don't get the data.

Reply:
@bardliao, it is an error for sure. Some messages _must_ have data in payload, like the RESOURCE_EVENT and the will be added MODULE_NOTIFICATION.

What I'm not sure is that will it ever happen? I think not, but have not checked it.